Added Product List to AccommodationSearchParameters

SamJaarsma opened the pull request:
Added Product List to AccommodationSearchParameters which was accidentally removed
Fixed a typo in the description of the AccommodationSearchParameters diagram

SamJaarsma committed:
added product list

havan commented on proto/cmp/types/v1alpha1/product_list.proto:
```diff
@@ -0,0 +1,22 @@
+//
+// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/types/v1alpha1/product-list.proto.dot.xs.svg)
+// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/types/v1alpha1/product-list.proto.dot.svg)
+message ProductList {
+  string product_code = 1;
+  ProductCodeType product_code_type = 2;
+}
```
In my opinion, this message type's naming is misleading. It suggests an already plural type Product**List**, but the actual message type is not. I suggest we change the naming to reflect that

```suggestion
message ProductCode {
  string code = 1;
  ProductCodeType type = 2;
}
```

SamJaarsma replied:
Agreed and implemented

havan commented on proto/cmp/types/v1alpha1/product_list.proto:
```diff
@@ -0,0 +1,22 @@
+//
+// This is being used in requests to specify the a list of products to be included in the reponse
+//
+// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/types/v1alpha1/product-list.proto.dot.xs.svg)
+// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/types/v1alpha1/product-list.proto.dot.svg)
```
The file name here is not correct, it should use the exact name of the `.proto` file. In this case it should be `product_list.proto`. **But** because I suggested to change the message type name to `ProductCode`, I will also suggest to change the file name to `product_code.proto`

```suggestion
// ![Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/types/v1alpha1/product_code.proto.dot.xs.svg)
// [Open Message Diagram](https://storage.googleapis.com/docs-cmp-files/diagrams/proto/cmp/types/v1alpha1/product_code.proto.dot.svg)
```

SamJaarsma replied:
Agreed and implemented

havan commented on proto/cmp/services/accommodation/v1alpha1/search_parameters_types.proto:
```diff
@@ -39,8 +40,10 @@ message AccommodationSearchParameters {
+
+  // Product code list
+  // Here a list of property codes would be used
+  repeated cmp.types.v1alpha1.ProductList product_list = 9;
```
To be in line with the other comments this part should also be updated as this:

```suggestion
  repeated cmp.types.v1alpha1.ProductCode product_codes = 9;
```

SamJaarsma replied:
Agreed and implemented

havan reviewed (changes requested):
I've made a few suggestions and I also have a general concern: Is this message type a **general one** that will be used through different verticals **or is it only related to Accommodation**?

SamJaarsma commented:
> I've made a few suggestions 
All agreed and implemented

> Is this message type a **general one** that will be used through different verticals **or is it only related to Accommodation**?

ProductList can be used in any message type, if OTAs have the products of several suppliers mapped to their own product codes and descriptions. Also amended Transport and Activity Message Types.

A ProductList with one entry can be used to query for a specific hotel, flight, transfer, rent-a-car,... when the initial search is done from a cache file.

I also changed the filename from types/product_list.proto to types/product_code.proto

SamJaarsma committed:
review comments implemented

SamJaarsma committed:
Implemented product_code message type instead of string in transport and activities

SamJaarsma committed:
improved single room vs multi room explanation

havan reviewed (approved):
LGTM
